Update Build Script

fynnwilliam opened the pull request:
Updated the minimum reversion of CMake and also fixed the bash script for compiling.

fynnwilliam committed:
Compatibility with older versions is not required

Compatibility with CMake < 2.8.12 will be removed from a future version
of CMake.

fynnwilliam committed:
Move into the build directory to proceed.

The script was missing a command - "cd build" as the third step.

goncaloalmeida reviewed (changes requested):
why add a max cmake version?

fynnwilliam committed:
CMake older than 2.8.12 is deprecated.

Reference here: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#command:cmake_minimum_required

fynnwilliam commented:
My bad, Goncalo. Updated the minimum value to be '2.8.12' as the older versions at deprecated.

Thank you.

fynnwilliam reviewed (commented):
Update cmake minimum version to 2.8.12

fynnwilliam committed:
Merge branch 'COVESA:master' into master

fynnwilliam committed:
Update CMake minimum version

I thought this has been merged long ago. At the moment it is safer to set minimum version to 3.5
